src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -382,8 +383,10 @@ private void OnChildItemCompleted(object sender, EventArgs e)
         /// </summary>
         private void OnAllChildItemsCompleted()
         {
-            var teardown = new OneTimeTearDownWorkItem(this);
-            Context.Dispatcher.Dispatch(teardown);
+            if (Context.ExecutionStatus == TestExecutionStatus.AbortRequested)
```

Comment:
If we're aborting, the one time teardown won't run so we don't enqueue it. Instead, we just mark the item as complete.

Come to think, maybe this should be a comment in the code.